applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1006,87 +1052,106 @@ static void stream_configured_cb(struct bt_bap_stream *stream,
 
 	if (dir == BT_AUDIO_DIR_SINK) {
 		/* NOTE: The string below is used by the Nordic CI system */
-		LOG_INF("%s sink stream configured", unicast_servers[device_index].ch_name);
-		le_audio_print_codec(
-			unicast_servers[device_index].cap_sink_stream.bap_stream.codec_cfg, dir);
+		LOG_INF("%s sink stream configured",
+			unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].ch_name);
+		le_audio_print_codec(unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx]
+					     .cap_sink_stream.bap_stream.codec_cfg,
+				     dir);
 	} else if (dir == BT_AUDIO_DIR_SOURCE) {
-		LOG_INF("%s source stream configured", unicast_servers[device_index].ch_name);
-		le_audio_print_codec(
-			unicast_servers[device_index].cap_source_stream.bap_stream.codec_cfg, dir);
+		LOG_INF("%s source stream configured",
+			unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].ch_name);
+		le_audio_print_codec(unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx]
+					     .cap_source_stream.bap_stream.codec_cfg,
+				     dir);
 	} else {
 		LOG_ERR("Endpoint direction not recognized: %d", dir);
 		return;
 	}
-	LOG_DBG("Configured Stream info: %s, %p, dir %d", unicast_servers[device_index].ch_name,
-		(void *)stream, dir);
+	LOG_DBG("Configured Stream info: %s, %p, dir %d",
+		unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].ch_name, (void *)stream, dir);
 
-	ret = device_pres_delay_find(device_index, &new_pres_dly_us);
+	ret = device_pres_delay_find(stream_idx, &new_pres_dly_us);
 	if (ret) {
 		LOG_ERR("Cannot get a valid presentation delay");
 		return;
 	}
 
-	if (unicast_servers[device_index].waiting_for_source_disc) {
+	if (unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].waiting_for_source_disc) {
 		return;
 	}
 
-	if (le_audio_ep_state_check(unicast_servers[device_index].cap_sink_stream.bap_stream.ep,
-				    BT_BAP_EP_STATE_CODEC_CONFIGURED)) {
-		for (int i = 0; i < ARRAY_SIZE(unicast_servers); i++) {
-			if (i != device_index && unicast_servers[i].device_conn != NULL) {
-				ret = update_cap_sink_stream_qos(&unicast_servers[i],
-								 new_pres_dly_us);
+	if (le_audio_ep_state_check(
+		    unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].cap_sink_stream.bap_stream.ep,
+		    BT_BAP_EP_STATE_CODEC_CONFIGURED)) {
+		for (int i = 0; i < ARRAY_SIZE(unicast_servers[lvl1_idx][lvl2_idx]); i++) {
+			if (i != lvl3_idx &&
+			    unicast_servers[lvl1_idx][lvl2_idx][i].device_conn != NULL) {
+				ret = update_cap_sink_stream_qos(
+					&unicast_servers[lvl1_idx][lvl2_idx][i], new_pres_dly_us);
```

Comment:
Too many indentations here. Challenging to read.

Reply:
At the start of the function we do:
```C
struct le_audio_unicast_server unicast_server = unicast_servers[lvl1][lvl2][lvl3];
```
That would at least remove some of the text, but I don't think we can change much about this here because it is iterating over the unicast_servers in the same group and subgroup. 

The only way I see to avoid some of the indentations is doing something like:
```C
if (le_audio_ep_state_check(!unicast_server.cap_sink_stream.bap_stream.ep, BT_BAP_EP_STATE_CODEC_CONFIGURED) {
    goto below_if_statement;
```
and I know how much you like gotos

Reply:
Try a static function in the first if